### PR TITLE
Add status, upgrade, and some other minor things

### PR DESCRIPTION
**Pull request by troglobit**
This PR adds some new functionality for customers:

 - Add get-data RPC to read operational status for interface counters and firmware version
 - Add edit-config RPC and separate copy-config RPCs for setting config to specific datastore
 - Add upgrade support using the Infix install-bundle RPC
 - Add HTML server to ease device upgrade
 - Add Status bar to bottom, for showing status and errors
 - Refactor main view, Send button felt malplaced in the right pane, now under text area
 - Refactor default Send command handling to allow custom NETCONF RPC commands
 - Refactor right-hand pane a bit, add text entry labels
 - Add Find Device button to Connection Parameters, locates all NETCONF capable mDNS devices
 - Refactor main menu a bit, rename entries, simplify zoom, add common keybindings
 - Add Help menu with About, License, and Usage Guide (html viewer) dialog boxes

It's probably more, but that's a good overall summary.

### CODE REVIEW

**Comment by troglobit** [comment]
**Note:** I haven't added any GitHub action for building and packaging yet.  I played around a bit with it in https://github.com/troglobit/webby/ just to learn, but I never managed to get the Windows packaging to work (always were missing some DLL files).

Also, packaging was not possible in this project due to the missing profinet XML files.  I added a suggestion in the Usage Guide for keeping two such example files in-tree without worrying about revealing names of any end customer.

**Comment by troglobit** [comment]
Sorry, found (and fixed) an unhandled exception when I started testing on Windows[^1], hence the new late commits.  

[^1]: nothing professional, I got it working with "Git Bash" on Windows 10.

**Review by saba8814 — APPROVED** [review]
BRA JOBBAT!

**Comment by troglobit** [comment]
> BRA JOBBAT!

Tackar! 😅

File: main.py [hunk 1]
```diff
@@ -201,126 +1139,331 @@ def is_port_valid(self):
                 return True
             else:
                 return False
-        except:
+        except Exception:
```
Comment (mattiaswal) [inline main.py:1142]: Drop Exception:
except:

File: main.py [hunk 2]
```diff
@@ -201,126 +1139,331 @@ def is_port_valid(self):
...
+        self.cfg_mgr.save()
+        self.restart_file_server()
+        self.status("Settings saved and web server restarted.")
+
+    # PROFINET STATUS METHODS
+    def _full_path(self, relative_path):
+        """Local helper function to get full file path"""
+        try:
+            base_path = sys._MEIPASS
+        except Exception:
```
Comment (mattiaswal) [inline main.py:1162]: Drop Exception:
except:

File: main.py [hunk 3]
```diff
@@ -1,198 +1,1136 @@
...
+                                     'org.cinnamon.desktop.interface',
+                                     'gtk-theme'],
+                                    capture_output=True,
+                                    text=True)
+            if result.returncode == 0:
+                theme = result.stdout.strip().strip("'")
+                if 'dark' in theme.lower():
+                    return "Dark"
+                return "Light"
+        except Exception:
```
Comment (mattiaswal) [inline main.py:919]: Drop Exception:
except:

File: main.py [hunk 4]
```diff
@@ -1,198 +1,1136 @@
...
+                                     'org.gnome.desktop.interface',
+                                     'color-scheme'],
+                                    capture_output=True,
+                                    text=True)
+            if result.returncode == 0:
+                theme = result.stdout.strip().strip("'")
+                if 'dark' in theme.lower():
+                    return "Dark"
+                return "Light"
+        except Exception:
```
Comment (mattiaswal) [inline main.py:934]: Drop Exception:
except:

**Review by mattiaswal — CHANGES_REQUESTED** [review]
Great work, just minor things. 

except Exception, 
Exception is redundant, since only except will catch all exceptions

**Comment by troglobit** [comment]
> except Exception, Exception is redundant, since only except will catch all exceptions

Ah of course, will fix asap, asap!

**Review by mattiaswal — APPROVED** [review]

**Pull request merged**